authorization/v2/uniformResourceIdentifier_test.go

```diff
@@ -224,3 +215,173 @@ func Test_UnmarshalJSON__should_return_correct_URI(t *testing.T) {
+		target UniformResourceIdentifier
+	}{
+		{
+			name: "only path",
+			source: UniformResourceIdentifier{
+				path: "hs:hs_auth:api:v2",
+			},
+			target: UniformResourceIdentifier{
+				path: "hs:hs_auth:api:v2:provide_access_to_uri",
+			},
+		},
+		{
+			name: "path and arguments",
+			source: UniformResourceIdentifier{
+				path:      "hs:hs_auth:api:v2:provide_access_to_uri",
+				arguments: map[string]string{"allowed_uri": "hs:hs_application:*"},
+			},
+			target: UniformResourceIdentifier{
+				path:      "hs:hs_auth:api:v2:provide_access_to_uri",
+				arguments: map[string]string{"allowed_uri": "hs:hs_application:checkin:*"},
+			},
+		},
+		{
+			name: "path, arguments and metadata",
+			source: UniformResourceIdentifier{
+				path:      "hs:hs_auth:api:v2:provide_access_to_uri",
+				arguments: map[string]string{"allowed_uri": "hs:hs_application:*"},
+				metadata:  map[string]string{"until": "21392103"},
+			},
+			target: UniformResourceIdentifier{
+				path:      "hs:hs_auth:api:v2:provide_access_to_uri",
+				arguments: map[string]string{"allowed_uri": "hs:hs_application:checkin:*"},
+				metadata:  map[string]string{"until": "21392103"},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			valid := tt.source.isSubsetOf(tt.target)
+
+			assert.Equal(t, valid, true)
+		})
+	}
+}
+
+func Test_isSubsetOf__should_return_false_with_source_not_in_target_set(t *testing.T) {
+	tests := []struct {
+		name   string
+		source UniformResourceIdentifier
+		target UniformResourceIdentifier
+	}{
+		{
+			name: "only path",
+			source: UniformResourceIdentifier{
+				path: "hs:hs_application:user:@me",
+			},
+			target: UniformResourceIdentifier{
+				path: "hs:hs_application:user",
+			},
+		},
+		{
+			name: "only path with same number of components",
+			source: UniformResourceIdentifier{
+				path: "hs:hs_auth:api:v2:GetUser",
+			},
+			target: UniformResourceIdentifier{
+				path: "hs:hs_auth:api:v2:GetUsers",
+			},
+		},
+		{
+			name: "path and arguments",
+			source: UniformResourceIdentifier{
+				path:      "hs:hs_auth:api:v2:provide_access_to_uri",
+				arguments: map[string]string{"allowed_uri": "hs:hs_application:checkin:*"},
+				metadata:  nil,
+			},
+			target: UniformResourceIdentifier{
+				path:      "hs:hs_auth:api:v2:provide_access_to_uri",
+				arguments: map[string]string{"allowed_uri": "hs:hs_application:*"},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			valid := tt.source.isSubsetOf(tt.target)
+
+			assert.Equal(t, valid, false)
+		})
+	}
+}
+
+func Test_isSubsetOf__should_return_false_when_target_doesnt_contain_source_arguments(t *testing.T) {
+	testSource := UniformResourceIdentifier{
+		path:      "hs:hs_auth",
+		arguments: map[string]string{"test": "1"},
+	}
+	testTarget := UniformResourceIdentifier{
+		path: "hs:hs_auth",
+	}
+
+	valid := testSource.isSubsetOf(testTarget)
+	assert.Equal(t, valid, false)
```

Comment:
This is not correct.

`hs:hs_auth?test=1` is a subset of `hs:hs_auth`. The `test=1` says that `hs:hs_auth?test=1` identifies all resources identified by `hs:hs_auth` as long as the value of `test` is `1`

Reply:
Good point. I'll fix those tests!